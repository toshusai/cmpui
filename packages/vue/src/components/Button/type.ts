export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
