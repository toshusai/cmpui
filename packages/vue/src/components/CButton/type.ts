export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
