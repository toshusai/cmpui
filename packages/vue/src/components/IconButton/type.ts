export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
