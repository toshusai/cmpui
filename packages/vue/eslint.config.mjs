"use strict";
import pluginVue from "eslint-plugin-vue";
import eslintConfigPrettier from "eslint-config-prettier";
import tseslint from "typescript-eslint";

export default [
  ...tseslint.configs.recommended,
  ...pluginVue.configs["flat/recommended"],
  {
    ignores: ["dist/*", "types/*"],
  },
  {
    rules: {
      "vue/multi-word-component-names": "off",
      "no-unused-vars": "off",
      "@typescript-eslint/no-unused-vars": ["error"],
    },
    languageOptions: {
      parserOptions: { parser: tseslint.parser, sourceType: "module" },
    },
  },
  eslintConfigPrettier,
];
