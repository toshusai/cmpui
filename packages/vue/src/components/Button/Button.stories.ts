import type { Meta, StoryObj } from "@storybook/vue3";
import Button from "./Button.vue";

const meta = {
  component: Button,
  tags: ["autodocs"],
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    default: "Button",
    variant: "primary",
  },
};

export const Secondary: Story = {
  args: {
    default: "Button",
    variant: "secondary",
  },
};

export const Small: Story = {
  args: {
    default: "Button",
    size: "S",
  },
};

export const Disabled: Story = {
  args: {
    default: "Button",
    disabled: true,
  },
};
