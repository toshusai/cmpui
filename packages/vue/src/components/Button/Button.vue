<script setup lang="ts">
import { ButtonProps } from "./type";

withDefaults(defineProps<ButtonProps>(), {
  variant: "primary",
  size: "M",
  disabled: false,
});
</script>

<template>
  <button
    class="cmpui_button"
    :data-variant="variant"
    :data-size="size"
    :disabled="disabled"
  >
    <slot></slot>
  </button>
</template>
