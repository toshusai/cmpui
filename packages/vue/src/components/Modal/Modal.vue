<script setup lang="ts">
import { ref, watch } from "vue";
import { focusTrap } from "../../lib/dialog";
import { ariaHiddenOthers } from "../../lib/dialog/ariaHiddenOthers";
import { lockScroll } from "../../lib/dialog/lockScroll";
import IconButton from "../IconButton/IconButton.vue";
import { useId } from "../../lib/useId";

const el = ref<HTMLElement | null>(null);

const props = defineProps<{
  show: boolean;
  title: string;
}>();

const emit = defineEmits<{
  (_e: "close"): void;
}>();

let cleanUp = () => {};

watch(
  () => el.value,
  (el) => {
    if (el) {
      const cleanUpFocusTrap = focusTrap(el);
      const cleanUpLockScroll = lockScroll();
      const cleanUpAriaHiddenOthers = ariaHiddenOthers(el);
      cleanUp = () => {
        cleanUpFocusTrap();
        cleanUpLockScroll();
        cleanUpAriaHiddenOthers();
      };
    }
  },
);

watch(
  () => props.show,
  (show) => {
    if (!show) cleanUp();
  },
);
const id = useId();
</script>

<template>
  <Transition name="cmpui_modal_transition">
    <Teleport v-if="show" to="body">
      <div
        ref="el"
        class="cmpui_modal__overlay"
        aria-modal="true"
        :aria-labelledby="id"
        @keydown.esc="emit('close')"
      >
        <div role="dialog" class="cmpui_float-box__root cmpui_modal__root">
          <div class="cmpui_modal__header">
            <div :id="id" class="cmpui_modal__title">{{ title }}</div>
            <IconButton
              class="cmpui_modal__close-button"
              size="S"
              @click="emit('close')"
            >
            </IconButton>
          </div>
          <slot></slot>
        </div>
      </div>
    </Teleport>
  </Transition>
</template>
