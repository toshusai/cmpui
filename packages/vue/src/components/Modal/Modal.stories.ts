import type { Meta, StoryObj } from "@storybook/vue3";
import { ref } from "vue";
import Modal from "./Modal.vue";
import Button from "../Button/Button.vue";

const meta = {
  title: "Components/Modal",
  component: Modal,
  tags: ["autodocs"],
} satisfies Meta<typeof Modal>;

export default meta;
type Story = StoryObj<typeof meta>;

const html = String.raw;
export const Default: Story = {
  render: () => ({
    setup() {
      const show = ref(false);
      return {
        show,
      };
    },
    components: { Modal, Button },
    template: html`
      <Modal :show="show" title="Confirm" @close="show = false">
        <div style="padding: 8px">
          <p style="font-size: 14px; margin: 0">
            Are you sure you want to delete this item?
          </p>
          <div
            style="display: flex; gap: 8px; justify-content: right; margin-top: 8px"
          >
            <Button size="S" variant="secondary" @click="show = false">
              Cancel
            </Button>
            <Button size="S" @click="show = false">OK</Button>
          </div>
        </div>
      </Modal>
      <Button @click="show = true">Open</Button>
    `,
  }),
};
