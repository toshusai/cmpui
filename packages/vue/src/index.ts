import Button from "./components/Button/Button.vue";

export { Button };
