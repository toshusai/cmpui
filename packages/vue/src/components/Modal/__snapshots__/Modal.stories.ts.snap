// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Modal Default smoke-test 1`] = `
<button class="cmpui_button"
        data-variant="primary"
        data-size="M"
>
  Open
</button>
`;
